### DISCUSSION TITLE
ScriptReference/Unity.Jobs.IJobExtensions.Schedule

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#157] [Notes]
# ScriptReference/Unity.Jobs.IJobExtensions.Schedule 29cc9053

https://docs.unity3d.com/ScriptReference/Unity.Jobs.IJobExtensions.Schedule.html

**Comment by vertxxyz** [comment]
Job execution doesn't start until either [`JobHandle.ScheduleBatchedJobs`](https://docs.unity3d.com/ScriptReference/Unity.Jobs.JobHandle.ScheduleBatchedJobs.html) or [`Complete`](https://docs.unity3d.com/ScriptReference/Unity.Jobs.JobHandle.Complete.html) is called[^1].
Unity's subsystems often use jobs themselves, calling `ScheduleBatchedJobs` regularly, so Scheduled jobs may kick off without you calling either. Just be aware you should probably call it yourself when you have your jobs ready.
[^1]: In the [Entity Component System](https://docs.unity3d.com/Packages/com.unity.entities@latest) the batch is implicitly flushed for you, so calling `ScheduleBatchedJobs` is not necessary.
